# Rosalind Bioinformatics Solutions

Solutions to problems on Rosalind, a platform for learning bioinformatics and programming through problem solving. My solutions are worked in Python 3, using tools including Biopython and NumPy. 

**Rosalind: http://rosalind.info**

**My profile: http://rosalind.info/users/anna22/**

| PROBLEM | NAME                  | INFO     | SOLUTION |
| --------|-----------------------|----------|----------|
| DNA     | Counting DNA Nucleotides|[Info](http://rosalind.info/problems/dna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/DNA_Counting%20Nucleotides.py)|
| RNA     | Transcribing DNA into RNA|[Info](http://rosalind.info/problems/rna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RNA_Transcribing%20DNA%20into%20RNA.py)|
| REVC    | Complementing a Strand of DNA|[Info](http://rosalind.info/problems/revc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/REVC_Complementing%20a%20Strand%20of%20DNA.py)|
| FIB     | Rabbits and Recurrence Relations|[Info](http://rosalind.info/problems/fib/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FIB_Rabbits%20and%20Recurrence%20Relations.py)|
| GC      | Computing GC Content|[Info](http://rosalind.info/problems/gc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GC_Computing%20GC%20Content.py)|
| HAMM    | Counting Point Mutations|[Info](http://rosalind.info/problems/hamm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/HAMM_Counting%20point%20mutations.py)|
| IPRB    | Mendel's First Law|[Info](http://rosalind.info/problems/iprb/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/IPRB_Mendel's%20first%20law.py)|
| PROT    | Translating RNA into Protein|[Info](http://rosalind.info/problems/prot/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PROT_Translating%20RNA%20into%20Protein.py)|
| SUBS    | Finding a Motif in DNA|[Info](http://rosalind.info/problems/subs/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SUBS_Finding%20a%20Motif%20in%20DNA.py)|
| CONS    | Consensus and Profile|[Info](http://rosalind.info/problems/cons/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CONS_Consensus%20and%20profile.py)|
| FIBD    | Mortal Fibonacci Rabbits|[Info](http://rosalind.info/problems/fibd/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FIBD_Mortal%20Fibonacci%20Rabbits.py)|
| GRPH    | Overlap Graphs|[Info](http://rosalind.info/problems/grph/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GRPH_Overlap%20graphs.py)|
| IEV     | Calculating Expected Offspring|[Info](http://rosalind.info/problems/iev/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/IEV_Calculating%20expected%20offspring.py)|
| LCSM    | Finding a Shared Motif|[Info](http://rosalind.info/problems/lcsm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LCSM_Finding%20a%20shared%20motif.py)|
| LIA     | Independent Alleles|[Info](http://rosalind.info/problems/lia/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LIA_Independent%20alleles.py)|
| MPRT    | Finding a Protein Motif|[Info](http://rosalind.info/problems/mprt/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MPRT_Finding%20a%20protein%20motif.py)|
| MRNA    | Inferring mRNA from Protein|[Info](http://rosalind.info/problems/mrna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MRNA_Inferring%20mRNA%20from%20Protein.py)|
| ORF     | Open Reading Frames|[Info](http://rosalind.info/problems/orf/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/ORF_Open%20Reading%20Frames.py)|
| PERM    | Enumerating Gene Orders|[Info](http://rosalind.info/problems/perm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PERM_Enumerating%20gene%20orders.py)|
| PRTM    | Calculating Protein Mass|[Info](http://rosalind.info/problems/prtm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PRTM_Calculating%20Protein%20Mass.py)|
| REVP    | Locating Restriction Sites|[Info](http://rosalind.info/problems/revp/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/REVP_Locating%20Restriction%20Sites.py)|
| SPLC    | RNA Splicing|[Info](http://rosalind.info/problems/splc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SPLC_RNA%20splicing.py)|
| LEXF    | Enumerating k-mers Lexicographically|[Info](http://rosalind.info/problems/lexf/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LEXF_Enumerating%20k-mers%20lexicographically.py)|
| LGIS    | Longest Increasing Subsequence|[Info](http://rosalind.info/problems/lgis/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LGIS_Longest%20Increasing%20Subsequence.py)|
| LONG    | Genome Assembly as Shortest Superstring|[Info](http://rosalind.info/problems/long/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LONG_Genome%20Assembly%20as%20Shortest%20Superstring.py)|
| PMCH    | Perfect Matchings and RNA Secondary Structures|[Info](http://rosalind.info/problems/pmch/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PMCH_Perfect%20Matchings%20and%20RNA%20Secondary%20Structures.py)|
| PPER    | Partial Permutations|[Info](http://rosalind.info/problems/pper/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PPER_Partial%20permutations.py)|
| PROB    | Introduction to Random Strings|[Info](http://rosalind.info/problems/prob/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PROB_Introduction%20to%20Random%20Strings.py)|
| SIGN    | Enumerating Oriented Gene Orderings|[Info](http://rosalind.info/problems/sign/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SIGN_Enumerating%20oriented%20gene%20orderings.py)|
| SSEQ    | Finding a Spliced Motif|[Info](http://rosalind.info/problems/sseq/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SSEQ_Finding%20a%20spliced%20motif.py)|
| TRAN    | Transitions and Transversions|[Info](http://rosalind.info/problems/tran/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/TRAN_Transitions%20and%20transversions.py)|
| TREE    | Completing a Tree|[Info](http://rosalind.info/problems/tree/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/TREE_Completing%20a%20tree.py)|
| CAT     | Catalan Numbers and RNA Secondary Structures|[Info](http://rosalind.info/problems/cat/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CAT_Catalan%20Numbers%20and%20RNA%20Secondary%20Structures.py)|
| CORR    | Error Correction in Reads|[Info](http://rosalind.info/problems/corr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CORR_Error%20Correction%20in%20Reads.py)|
| INOD    | Counting Phylogenetic Ancestors|[Info](http://rosalind.info/problems/inod/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/INOD_Counting%20Phylogenetic%20Ancestors.py)|
| KMER    | k-Mer Composition|[Info](http://rosalind.info/problems/kmer/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/KMER_k-Mer%20Composition.py)|
| KMP     | Speeding Up Motif Finding|[Info](http://rosalind.info/problems/kmp/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/KMP_Speeding%20Up%20Motif%20Finding.py)|
| LCSQ    | Finding a Shared Spliced Motif|[Info](http://rosalind.info/problems/lcsq/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LCSQ_Finding%20a%20Shared%20Spliced%20Motif.py)|
| LEXV    | Ordering Strings of Varying Length Lexicographically|[Info](http://rosalind.info/problems/lexv/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LEXV_Ordering%20Strings%20of%20Varying%20Length%20Lexicographically.py)|
| MMCH    | Maximum Matchings and RNA Secondary Structures|[Info](http://rosalind.info/problems/mmch/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MMCH_Maximum%20Matchings%20and%20RNA%20Secondary%20Structures.py)|
| PDST    | Creating a Distance Matrix|[Info](http://rosalind.info/problems/pdst/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PDST_Creating%20a%20Distance%20Matrix.py)|
| RSTR    | Matching Random Motifs|[Info](http://rosalind.info/problems/rstr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RSTR_Matching%20Random%20Motifs.py)|
| SSET    | Counting Subsets|[Info](http://rosalind.info/problems/sset/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SSET_Counting%20subsets.py)|
| ASPC    | Introduction to Alternative Splicing|[Info](http://rosalind.info/problems/aspc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/ASPC_Introduction%20to%20Alternative%20Splicing.py)|
| EDIT    | Edit Distance|[Info](http://rosalind.info/problems/edit/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/EDIT_Edit%20Distance.py)|
| EVAL    | Expected Number of Restriction Sites|[Info](http://rosalind.info/problems/eval/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/EVAL_Expected%20Number%20of%20Restriction%20Sites.py)|
| MOTZ    | Motzkin Numbers and RNA Secondary Structures|[Info](http://rosalind.info/problems/motz/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MOTZ_Motzkin%20Numbers%20and%20RNA%20Secondary%20Structures.py)|
| NWCK    | Distances in Trees|[Info](http://rosalind.info/problems/nwck/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/NWCK_Distance%20in%20Trees.py)|
| SCSP    | Interleaving Two Motifs|[Info](http://rosalind.info/problems/scsp/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SCSP_Interleaving%20Two%20Motifs.py)|
| SETO    | Introduction to Set Operations|[Info](http://rosalind.info/problems/seto/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SETO_Introduction%20to%20Set%20Operations.py)|
| SPEC    | Inferring Protein from Spectrum|[Info](http://rosalind.info/problems/spec/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SPEC_Inferring%20Protein%20from%20Spectrum.py)|
| TRIE    | Introduction to Pattern Matching|[Info](http://rosalind.info/problems/trie/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/TRIE_Introduction%20to%20Pattern%20Matching.py)|
| CONV    | Comparing Spectra with the Spectral Convolution|[Info](http://rosalind.info/problems/conv/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CONV_Comparing%20Spectra%20with%20the%20Spectral%20Convolution.py)|
| CTBL    | Creating a Character Table|[Info](http://rosalind.info/problems/ctbl/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CTBL_Creating%20a%20Character%20Table.py)|
| DBRU    | Constructing a De Bruijn Graph|[Info](http://rosalind.info/problems/dbru/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/DBRU_Constructing%20a%20De%20Bruijn%20Graph.py)|
| EDTA    | Edit Distance Alignment|[Info](http://rosalind.info/problems/edta/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/EDTA_Edit%20Distance%20Alignment.py)|
| FULL    | Inferring Peptide from Full Spectrum|[Info](http://rosalind.info/problems/full/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FULL_Inferring%20Peptide%20from%20Full%20Spectrum.py)|
| INDC    | Independent Segregation of Chromosomes|[Info](http://rosalind.info/problems/indc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/INDC_Independent%20Segregation%20of%20Chromosomes.py)|
| LREP    | Finding the Longest Multiple Repeat|[Info](http://rosalind.info/problems/lrep/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LREP_Finding%20the%20Longest%20Multiple%20Repeat.py)|
| NKEW    | Newick Format with Edge Weights|[Info](http://rosalind.info/problems/nkew/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/NKEW_Newick%20Format%20with%20Edge%20Weights.py)|
| RNAS    | Wobble Bonding and RNA Secondary Structures|[Info](http://rosalind.info/problems/rnas/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RNAS_Wobble%20Bonding%20and%20RNA%20Secondary%20Structures.py)|
| AFRQ    | Counting Disease Carriers|[Info](http://rosalind.info/problems/afrq/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/AFRQ_Counting%20Disease%20Carriers.py)|
| CSTR    | Creating a Character Table from Genetic Strings|[Info](http://rosalind.info/problems/cstr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CSTR_Creating%20a%20Character%20Table%20from%20Genetic%20Strings.py)|
| CUNR    | Counting Unrooted Binary Trees|[Info](http://rosalind.info/problems/cunr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CUNR_Counting%20Unrooted%20Binary%20Trees.py)|
| GLOB    | Global Alignment with Scoring Matrix|[Info](http://rosalind.info/problems/glob/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GLOB_Global%20Alignment%20with%20Scoring%20Matrix.py)|
| PCOV    | Genome Assembly with Perfect Coverage|[Info](http://rosalind.info/problems/pcov/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PCOV_Genome%20Assembly%20with%20Perfect%20Coverage.py)|
| PRSM    | Matching a Spectrum to a Protein|[Info](http://rosalind.info/problems/prsm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PRSM_Matching%20a%20Spectrum%20to%20a%20Protein.py)|
| SGRA    | Using the Spectrum Graph to Infer Peptides|[Info](http://rosalind.info/problems/sgra/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SGRA_Using%20the%20Spectrum%20Graph%20to%20Infer%20Peptides.py)|
| GASM    | Genome Assembly Using Reads|[Info](http://rosalind.info/problems/gasm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GASM_Genome%20Assembly%20Using%20Reads.py)|
| MEND    | Inferring Genotype from a Pedigree|[Info](http://rosalind.info/problems/mend/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MEND_Inferring%20Genotype%20from%20a%20Pedigree.py)|
| PDPL    | Creating a Restriction Map|[Info](http://rosalind.info/problems/pdpl/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PDPL_Creating%20a%20Restriction%20Map.py)|
| ROOT    | Counting Rooted Binary Trees|[Info](http://rosalind.info/problems/root/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/ROOT_Counting%20Rooted%20Binary%20Trees.py)|
| SEXL    | Sex-Linked Inheritance|[Info](http://rosalind.info/problems/sexl/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SEXL_Sex-Linked%20Inheritance.py)|
| WFMD    | The Wright-Fisher Model of Genetic Drift|[Info](http://rosalind.info/problems/wfmd/)|
| INI     | Introduction to the Bioinformatics Armory|[Info](http://rosalind.info/problems/ini/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/INI_Introduction%20to%20the%20Bioinformatics%20Armory.py)|
| DBPR    | Introduction to Protein Databases|[Info](http://rosalind.info/problems/dbpr/)|
| GBK     | GenBank Introduction|[Info](http://rosalind.info/problems/gbk/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GBK_GenBank%20Introduction.py)|
| FRMT    | Data Formats|[Info](http://rosalind.info/problems/frmt/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FRMT_Data%20Formats.py)|
| NEED    | Pairwise Global Alignment|[Info](http://rosalind.info/problems/need/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/NEED_Pairwise%20Global%20Alignment.py)|
| TFSQ    | FASTQ Format Introduction|[Info](http://rosalind.info/problems/tfsq/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/TFSQ_FASTQ%20Format%20Introduction.py)|
| PHRE    | Read Quality Distribution|[Info](http://rosalind.info/problems/phre/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PHRE_Read%20Quality%20Distribution.py)|
| PTRA    | Protein Translation|[Info](http://rosalind.info/problems/ptra/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PTRA_Protein%20Translation.py)|
| RVCO    | Complementing a Strand of DNA|[Info](http://rosalind.info/problems/rvco/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RVCO_Complementing%20a%20Strand%20of%20DNA.py)|
| FILT    | Read Filtration by Quality|[Info](http://rosalind.info/problems/filt/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FILT_Read%20Filtration%20by%20Quality.py)|
| SUBO    | Suboptimal Local Alignment|[Info](http://rosalind.info/problems/subo/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SUBO_Suboptimal%20Local%20Alignment.py)|
| BPHR    | Base Quality Distribution|[Info](http://rosalind.info/problems/bphr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/BPHR_Base%20Quality%20Distribution.py)|
| ORFR    | Finding Genes with ORFs|[Info](http://rosalind.info/problems/orfr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/ORFR_Finding%20Genes%20with%20ORFs.py)|
| BFIL    | Base Filtration by Quality|[Info](http://rosalind.info/problems/bfil/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/BFIL_Base%20Filtration%20by%20Quality.py)|
