# Rosalind Bioinformatics Solutions

Solutions to problems on Rosalind, a platform for learning bioinformatics and programming through problem solving. My solutions are worked in Python 3.x, using tools including Biopython and NumPy. 

**Rosalind: http://rosalind.info**

**My profile: http://rosalind.info/users/anna22/**

| PROBLEM | NAME                  | INFO     | SOLUTION |
| --------|-----------------------|----------|----------|
| DNA     | Counting DNA Nucleotides|[Info](http://rosalind.info/problems/dna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/DNA_Counting%20Nucleotides.py)|
| RNA     | Transcribing DNA into RNA|[Info](http://rosalind.info/problems/rna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RNA_Transcribing%20DNA%20into%20RNA.py)|
| REVC    | Complementing a Strand of DNA|[Info](http://rosalind.info/problems/revc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/REVC_Complementing%20a%20Strand%20of%20DNA.py)|
| FIB     | Rabbits and Recurrence Relations|[Info](http://rosalind.info/problems/fib/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FIB_Rabbits%20and%20Recurrence%20Relations.py)|
| GC      | Computing GC Content|[Info](http://rosalind.info/problems/gc/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GC_Computing%20GC%20Content.py)|
| HAMM    | Counting Point Mutations|[Info](http://rosalind.info/problems/hamm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/HAMM_Counting%20point%20mutations.py)|
| IPRB    | Mendel's First Law|[Info](http://rosalind.info/problems/iprb/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/IPRB_Mendel's%20first%20law.py)|
| PROT    | Translating RNA into Protein|[Info](http://rosalind.info/problems/prot/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PROT_Translating%20RNA%20into%20Protein.py)|
| SUBS    | Finding a Motif in DNA|[Info](http://rosalind.info/problems/subs/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SUBS_Finding%20a%20Motif%20in%20DNA.py)|
| CONS    | Consensus and Profile|[Info](http://rosalind.info/problems/cons/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CONS_Consensus%20and%20profile.py)|
| FIBD    | Mortal Fibonacci Rabbits|[Info](http://rosalind.info/problems/fibd/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FIBD_Mortal%20Fibonacci%20Rabbits.py)|
| GRPH    | Overlap Graphs|[Info](http://rosalind.info/problems/grph/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/GRPH_Overlap%20graphs.py)|
| IEV     | Calculating Expected Offspring|[Info](http://rosalind.info/problems/iev/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/IEV_Calculating%20expected%20offspring.py)|
| LCSM    | Finding a Shared Motif|[Info](http://rosalind.info/problems/lcsm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LCSM_Finding%20a%20shared%20motif.py)|
| LIA     | Independent Alleles|[Info](http://rosalind.info/problems/lia/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LIA_Independent%20alleles.py)|
| MPRT    | Finding a Protein Motif|[Info](http://rosalind.info/problems/mprt/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MPRT_Finding%20a%20protein%20motif.py)|
| MRNA    | Inferring mRNA from Protein|[Info](http://rosalind.info/problems/mrna/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/MRNA_Inferring%20mRNA%20from%20Protein.py)|
| ORF     | Open Reading Frames|[Info](http://rosalind.info/problems/orf/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/ORF_Open%20Reading%20Frames.py)|
| PERM    | Enumerating Gene Orders|[Info](http://rosalind.info/problems/perm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PERM_Enumerating%20gene%20orders.py)|
| PRTM    | Calculating Protein Mass|[Info](http://rosalind.info/problems/prtm/)|
| REVP    | Locating Restriction Sites|[Info](http://rosalind.info/problems/revp/)|
| SPLC    | RNA Splicing|[Info](http://rosalind.info/problems/splc/)|
| LEXF    | Enumerating k-mers Lexicographically|[Info](http://rosalind.info/problems/lexf/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/LEXF_Enumerating%20k-mers%20lexicographically.py)|
| LGIS    | Longest Increasing Subsequence|[Info](http://rosalind.info/problems/lgis/)|
| LONG    | Genome Assembly as Shortest Superstring|[Info](http://rosalind.info/problems/long/)|
| PMCH    | Perfect Matchings and RNA Secondary Structures|[Info](http://rosalind.info/problems/pmch/)|
| PPER    | Partial Permutations|[Info](http://rosalind.info/problems/pper/)|
| PROB    | Introduction to Random Strings|[Info](http://rosalind.info/problems/prob/)|
| SIGN    | Enumerating Oriented Gene Orderings|[Info](http://rosalind.info/problems/sign/)|
| SSEQ    | Finding a Spliced Motif|[Info](http://rosalind.info/problems/sseq/)|
| TRAN    | Transitions and Transversions|[Info](http://rosalind.info/problems/tran/)|
| TREE    | Completing a Tree|[Info](http://rosalind.info/problems/tree/)|
| CAT     | Catalan Numbers and RNA Secondary Structures|[Info](http://rosalind.info/problems/cat/)|
| CORR    | Error Correction in Reads|[Info](http://rosalind.info/problems/corr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CORR_Error%20Correction%20in%20Reads.py)|
| INOD    | Counting Phylogenetic Ancestors|[Info](http://rosalind.info/problems/inod/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/INOD_Counting%20Phylogenetic%20Ancestors.py)|
| KMER    | k-Mer Composition|[Info](http://rosalind.info/problems/kmer/)|
| KMP     | Speeding Up Motif Finding|[Info](http://rosalind.info/problems/kmp/)|
| LCSQ    | Finding a Shared Spliced Motif|[Info](http://rosalind.info/problems/lcsq/)|
| LEXV    | Ordering Strings of Varying Length Lexicographically|[Info](http://rosalind.info/problems/lexv/)|
| MMCH    | Maximum Matchings and RNA Secondary Structures|[Info](http://rosalind.info/problems/mmch/)|
| PDST    | Creating a Distance Matrix|[Info](http://rosalind.info/problems/pdst/)|
| RSTR    | Matching Random Motifs|[Info](http://rosalind.info/problems/rstr/)|
| SSET    | Counting Subsets|[Info](http://rosalind.info/problems/sset/)|
| ASPC    | Introduction to Alternative Splicing|[Info](http://rosalind.info/problems/aspc/)|
| EDIT    | Edit Distance|[Info](http://rosalind.info/problems/edit/)|
| EVAL    | Expected Number of Restriction Sites|[Info](http://rosalind.info/problems/eval/)|
| MOTZ    | Motzkin Numbers and RNA Secondary Structures|[Info](http://rosalind.info/problems/motz/)|
| NWCK    | Distances in Trees|[Info](http://rosalind.info/problems/nwck/)|
| SCSP    | Interleaving Two Motifs|[Info](http://rosalind.info/problems/scsp/)|
| SETO    | Introduction to Set Operations|[Info](http://rosalind.info/problems/seto/)|
| SPEC    | Inferring Protein from Spectrum|[Info](http://rosalind.info/problems/spec/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SPEC_Inferring%20Protein%20from%20Spectrum.py)|
| TRIE    | Introduction to Pattern Matching|[Info](http://rosalind.info/problems/trie/)|
| CONV    | Comparing Spectra with the Spectral Convolution|[Info](http://rosalind.info/problems/conv/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CONV_Comparing%20Spectra%20with%20the%20Spectral%20Convolution.py)|
| CTBL    | Creating a Character Table|[Info](http://rosalind.info/problems/ctbl/)|
| DBRU    | Constructing a De Bruijn Graph|[Info](http://rosalind.info/problems/dbru/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/DBRU_Constructing%20a%20De%20Bruijn%20Graph.py)|
| EDTA    | Edit Distance Alignment|[Info](http://rosalind.info/problems/edta/)|
| FULL    | Inferring Peptide from Full Spectrum|[Info](http://rosalind.info/problems/full/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/FULL_Inferring%20Peptide%20from%20Full%20Spectrum.py)|
| NKEW    | Newick Format with Edge Weights|[Info](http://rosalind.info/problems/nkew/)|
| RNAS    | Wobble Bonding and RNA Secondary Structures|[Info](http://rosalind.info/problems/rnas/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/RNAS_Wobble%20Bonding%20and%20RNA%20Secondary%20Structures.py)|
| CSTR    | Creating a Character Table from Genetic Strings|[Info](http://rosalind.info/problems/cstr/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/CSTR_Creating%20a%20Character%20Table%20from%20Genetic%20Strings.py)|
| GLOB    | Global Alignment with Scoring Matrix|[Info](http://rosalind.info/problems/glob/)|
| PCOV    | Genome Assembly with Perfect Coverage|[Info](http://rosalind.info/problems/pcov/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PCOV_Genome%20Assembly%20with%20Perfect%20Coverage.py)|
| PRSM    | Matching a Spectrum to a Protein|[Info](http://rosalind.info/problems/prsm/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/PRSM_Matching%20a%20Spectrum%20to%20a%20Protein.py)|
| SGRA    | Using the Spectrum Graph to Infer Peptides|[Info](http://rosalind.info/problems/sgra/)|[Solution](https://github.com/agolikova/Rosalind-Bioinformatics-Solutions/blob/main/Code/SGRA_Using%20the%20Spectrum%20Graph%20to%20Infer%20Peptides.py)|
